### ISSUE TITLE
ObjectACL, BucketACL not quite correct behavior

### ISSUE THREAD

**Issue by evgeniiz321** [nspcc-dev/neofs-s3-gw#906]
https://docs.aws.amazon.com/AmazonS3/latest/userguide/managing-acls.html

```
Bucket and object permissions are independent of each other. An object does not inherit the permissions from its bucket. For example, if you create a bucket and grant write access to a user, you can't access that user’s objects unless the user explicitly grants you access.
```

1. Create bucket with public-read-write ACL
2. Create object1 with public-read-write ACL
3. Create object2 without ACL (by default, access should be allowed only by an object owner)
4. Try to get/put obj2 from a not owner - access allowed, but should be denied

Related to https://github.com/nspcc-dev/neofs-s3-gw/issues/904, but this one is a more general issue.

**Comment by smallhive** [comment]
I consider, `bucket with public-read-write` leads to `object2 with public-read-write` by default, according to the bucket rules

**Comment by roman-khimov** [comment]
Yeah, that's somewhat strange and I'd expect `object2` to be accessible. Is this behavior confirmed for AWS?

**Comment by roman-khimov** [comment]
Similar to #904.

**Comment by roman-khimov** [comment]
Obsoleted by #1275.
